students/ericstreit/lessons/lesson02/assignment/code/charges_calc.py

```diff
@@ -5,43 +5,135 @@
     parser.add_argument('-o', '--output', help='ouput JSON file', required=True)
-    parser.add_argument('-d', '--debugging', help='Debug Level', required=False)
+    parser.add_argument('-d', '--debug', help='Debug Level', type=int, default=0)
     return parser.parse_args()
 
 
 def load_rentals_file(filename):
+    """
+    Open the specified json file
+    """
+
     with open(filename) as file:
+        logging.debug(f'Opening {filename}')
         try:
             data = json.load(file)
-        except:
+        except Exception as except_error:
+            print(except_error)
+            logging.error(except_error)
             exit(0)
     return data
 
+
 def calculate_additional_fields(data):
-    for value in data.values():
+    """
+    Function that grabs the json data and creates addtional fields from the
+    calculations of existing data; Total Days, Total Price, Square Root Total
+     and Unit Cost
+    """
+
+    # for value in data.values():
+    logging.debug("Looping thru the json file to calculate the additional fields!")
+    for key, value in data.items():
+        # grab the item and values
+        logging.debug(f'Working with {key} {value}')
+        # start calculating and creating the additional fields
         try:
+            # a good place to check if the starting date is newer than the end
+            # date and correct it if possible
             rental_start = datetime.datetime.strptime(value['rental_start'], '%m/%d/%y')
             rental_end = datetime.datetime.strptime(value['rental_end'], '%m/%d/%y')
             value['total_days'] = (rental_end - rental_start).days
+        except Exception as except_error:
+            logging.error(f': {key} : Error calculating Total Days: {except_error}')
+            # pass
+        try:
             value['total_price'] = value['total_days'] * value['price_per_day']
+            logging.debug(f"Updating {value} total price to: "
+                            f"{value.get('total_price')}")
+        except Exception as except_error:
+            logging.error(f': {key} : Error calculating Total Price: {except_error}')
+            # pass
+        try:
             value['sqrt_total_price'] = math.sqrt(value['total_price'])
+            logging.debug(f"Updating {value} square root total price to: "
+                           f"{value.get('sqrt_total_price')}")
+        except Exception as except_error:
+            logging.error(f": {key} : Error calculating the Square Root Total "
+                            f"Price: {except_error}")
+            # pass
+        try:
             value['unit_cost'] = value['total_price'] / value['units_rented']
-        except:
-            exit(0)
+            logging.debug(f"Updating {value} unit cost to: {value.get('unit_cost')}")
+        except Exception as except_error:
+            logging.error(f': {key} : Error calculating Unit Cost: {except_error}')
+            # pass
```

Comment:
It is a lot of try/excepts, but that is what happens when cleaning data. Would be nice to catch the precise exception (like KeyError or whatever) rather than the generic Exception. This is because Exception catches everything, and sometime this is an unexpected error that you would prefer to throw, if you knew about it.